src/NUnitFramework/framework/XmlExtensions.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Xml;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// Contains extension methods that do not require a special <c>using</c> directive.
+    /// </summary>
+    internal static class XmlExtensions
+    {
+        // we want to write just the main element without XML declarations
+        internal static readonly XmlWriterSettings FragmentWriterSettings = new()
+        {
+            ConformanceLevel = ConformanceLevel.Fragment
+        };
+
+        /// <summary>
+        /// Checks that attribute value contains safe content and if not, escapes it.
+        /// </summary>
+        internal static void WriteAttributeStringSafe(this XmlWriter writer, string name, string value)
+        {
+            writer.WriteAttributeString(name, EscapeInvalidXmlCharacters(value));
+        }
+
+        /// <summary>
+        /// Checks that CDATA section contains safe content and if not, escapes it.
+        /// </summary>
+        internal static void WriteCDataSafe(this XmlWriter writer, string text)
+        {
+            if (text is null)
+                throw new ArgumentNullException(nameof(text));
+
+            text = EscapeInvalidXmlCharacters(text);
```

Comment:
question: I noticed we added the call to `EscapeInvalidXmlCharacters`. Were there any other changes as part of moving this into an extensions file? It's tough to tell for sure

Reply:
No other changes than ensuring the common fast path for "no need to encode anything".